include/fiction/utils/math_utils.hpp

```diff
@@ -117,6 +117,43 @@ determine_all_combinations_of_distributing_k_entities_on_n_positions(const std::
 
     return all_combinations;
 }
+/**
+ * This function computes the Cartesian product of a list of vectors. Each vector in the input list
+ * represents a dimension, and the function produces all possible combinations where each combination
+ * consists of one element from each dimension vector.
+ *
+ * @tparam VectorDataType The type of elements in the vectors.
+ * @param indices A vector of vectors, where each inner vector represents a dimension. The function
+ *                generates combinations using one element from each dimension vector.
+ * @return A vector of vectors, where each inner vector represents a combination of elements,
+ *         one from each dimension. The total number of combinations is the product of the sizes
+ *         of the input vectors.
+ */
+template <typename VectorDataType>
```

Comment:
@marcelwa I have added this function since it is quite useful, also for the displacement robustness domain simulation. Please double-check. Thanks!